rules/common/private/utils.bzl

```diff
@@ -117,6 +117,17 @@ def safe_name(value):
 def _short_path(file):
     return file.short_path
 
+# This propagates specific tags as execution requirements to be passed to an action
+# A fix to bazelbuild/bazel that will make this no longer necessary is underway; we can remove this once that's released and we've obtained it
+PROPAGATABLE_TAGS = ["no-remote", "no-cache", "no-sandbox", "no-remote-exec", "no-remote-cache"]
+
+def resolve_execution_reqs(ctx, base_exec_reqs):
+    exec_reqs = dict(base_exec_reqs)
+    for tag in ctx.attr.tags:
+        if tag in PROPAGATABLE_TAGS:
+            exec_reqs.update({tag: "1"})
```

Comment:
It may be better to do this in the opposite order, so attributes in the `base_exec_reqs` are not overridden by tags defined in the `BUILD` files. Though this may also be the better option. Not sure which one is more likely to cause unexpected behavior.

Reply:
Ah good point. That's what they do in `bazelbuild/bazel` too. I'll update